obfuscation.go

```diff
@@ -15,10 +15,12 @@ type OnionErrorEncrypter struct {
 // NewOnionErrorEncrypter creates new instance of the onion encrypter backed by
 // the passed router, with encryption to be doing using the passed
 // ephemeralKey.
-func NewOnionErrorEncrypter(router *Router,
-	ephemeralKey *btcec.PublicKey) (*OnionErrorEncrypter, error) {
+func NewOnionErrorEncrypter(router *Router, ephemeralKey,
+	blindingPoint *btcec.PublicKey) (*OnionErrorEncrypter, error) {
```

Comment:
Perhaps functional options should be used here? Since the typical path won't set these values, and as it it serves to break all the APIs somewhat unnecessarily.